### PR TITLE
Smack

### PR DESCRIPTION
**Pull request by HeroPD**
I changed smack 3.2.1 to smack4.1.8

### CODE REVIEW

**Comment by HeroPD** [comment]
I used XMPPTCPConnectionConfiguration.SecurityMode.disabled mode to connect server via smack, i am not sure what should i do about them

**Comment by wrooot** [comment]
Filed as https://issues.igniterealtime.org/browse/OF-1190

**Comment by lugreen** [comment]
支持

File: src/java/org/jivesoftware/webchat/ChatSession.java [hunk 1]
```diff
@@ -440,10 +474,15 @@ private void checkForEmptyRoom() {
         // See if we are the last one in the chat room.
         if (groupChat.getOccupantsCount() == 1) {
             // Leave the room.
-            groupChat.leave();
+            try {
+				groupChat.leave();
+			} catch (NotConnectedException e) {
+				// TODO Auto-generated catch block
```
Comment (guusdk) [inline src/java/org/jivesoftware/webchat/ChatSession.java]: Please don't ever do ``e.printStackTrace()``. Write the exception to the log instead.

File: src/java/org/jivesoftware/webchat/LoadTest.java [hunk 2]
```diff
@@ -90,78 +95,90 @@ public static void main(String[] args) {
...
-                public void invitationReceived(Connection conn, String room, String inviter, String reason, String password, Message message) {
-                    throw new UnsupportedOperationException();
-                }
-            });
-
-        }
-        catch (XMPPException e) {
-            e.printStackTrace();
-        }
+//        try {
```
Comment (guusdk) [inline src/java/org/jivesoftware/webchat/LoadTest.java]: What happened with this functinality?

Comment (HeroPD) [inline src/java/org/jivesoftware/webchat/LoadTest.java]: seems i forgot to convert them smack 4.1.8

File: src/java/org/jivesoftware/webchat/ChatSession.java (hunk 1 above)
Comment (HeroPD) [inline src/java/org/jivesoftware/webchat/ChatSession.java]: Okey got it i will write all exceptions to log

**Comment by HeroPD** [comment]
I added requested changed. Please review it

**Comment by akrherz** [comment]
@guusdk do the updated changes fix your review?

**Comment by guusdk** [comment]
Smack was updated (to a newer version) in #11

**Pull request closed without merging**